samples/sdl_gamecontroller/main.c

```diff
@@ -0,0 +1,105 @@
+#include <debug.h>
+#include <pbkit/pbkit.h>
+#include <hal/video.h>
+#include <hal/xbox.h>
+#include <assert.h>
+#include <SDL.h>
+#include <stdbool.h>
+
+int main() {
+    XVideoSetMode(640, 480, 32, REFRESH_DEFAULT);
+    int ret = pb_init();
+    if (ret != 0) {
+        XSleep(2000);
+        return -1;
+    }
+
+    pb_show_debug_screen();
+
+    SDL_InitSubSystem(SDL_INIT_GAMECONTROLLER);
+
+    int gameControllerIndex = 0;
+    SDL_GameController *gameController = SDL_GameControllerOpen(gameControllerIndex);
+    if (gameController == NULL) {
+        debugPrint("Could not open GameController %i, please plug in a\n", gameControllerIndex);
+        debugPrint("compatible controller into port 1. Exiting...\n");
+        SDL_QuitSubSystem(SDL_INIT_GAMECONTROLLER);
+        XSleep(2000);
+        pb_kill();
+        return -1;
+    }
+
+    while (true) {
+        // Fetch current GameController state
+        SDL_GameControllerUpdate();
+        debugClearScreen();
```

Comment:
The existing review comments, about checking wether `gameController` is still connected, seem to have been ignored / missed.

Consider https://wiki.libsdl.org/SDL_GameControllerGetAttached?highlight=%28%5CbCategoryGameController%5Cb%29%7C%28CategoryEnum%29